django/db/models/expressions.py

```diff
@@ -412,6 +432,12 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
     def refs_aggregate(self, existing_aggregates):
         return refs_aggregate(self.name.split(LOOKUP_SEP), existing_aggregates)
 
+    def asc(self):
```

Comment:
I think we need to rename CombinableMixin, or just stuff the asc() and desc() methods in the current CombinableMixin. The reason is that if I want to do some other wrapper expression than F() (say, SumIf that returns Sum(Case(When(...))) from resolve_expression, then I need to subclass CombinableMixin, and also add asc() and desc() methods. It would be better if we could just subclass CombinableMixin and write resolve_expression.

(Another case of needed methods is refs_aggregate() which we should be able to remove once where/having split is done in the compiling stage)